.github/workflows/test.yml

```diff
@@ -41,8 +45,8 @@ jobs:
         python-version: ${{ matrix.python-version }}
     - run: python -m pip install --upgrade pip wheel
     # python -m pip install .[test] is not used here to test minimum (faster),
-    # the cov workflow runs all tests
-    - run: python -m pip install -v . pytest
+    # the cov workflow runs all tests.
+    - run: python -m pip install -v . pytest ${{ matrix.installs }}
```

Comment:
This looks like it can be handled in pyproject.toml with environment markers. https://packaging.python.org/en/latest/specifications/dependency-specifiers/#environment-markers

Reply:
The goal is to try with the minimum and maximum supported numpy version. Everything from 1.21 to 2.0.0rc1 is supported on ll platforms, this is just a sampling for the tests.

Another way to do it would be to switch from using `pip` to `uv`, then `uv pip install -e . --resolution=lowest-direct` will install your minimum dependencies. I was actually using that locally in testing the patch. And `uv pip install . --prerelease` (regular pip supports this too) will install pre-releases, which would get numpy 2.0.0rc1.

Reply:
(If you do have a dependency that depends on the install environment, environment markers are exactly the way to go, but this is not that, it's just trying a selection of min and max deps for testing)

Reply:
Ok, thanks for clarifying, after I wrote that I also got some doubts whether I understood correctly what you are doing, which you confirm here. Thanks for the hint to uv, that seems pretty cool.